File: app/src/main/java/com/rowland/delivery/merchant/application/di/modules/ContextModule.java
package com.rowland.delivery.merchant.application.di.modules;

import android.app.Application;
import android.content.Context;

import com.rowland.delivery.merchant.application.di.scopes.ApplicationContextQualifier;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Rowland on 4/30/2018.
 */

@Module
public class ContextModule {

    private final Context context;

    public ContextModule(Context context) {
        this.context = context;
    }

    @Provides
    public Context providesContext() {
        return this.context;
    }
}

File: app/src/main/java/com/rowland/delivery/merchant/application/di/modules/ContextModule.kt
package com.rowland.delivery.merchant.application.di.modules

import android.content.Context
import dagger.Module
import dagger.Provides

/**
 * Created by Rowland on 4/30/2018.
 */

@Module
class ContextModule(private val context: Context) {

    @Provides
    fun providesContext(): Context {
        return this.context
    }
}
